src/components/SearchBox/SearchBox.tsx

```diff
@@ -1,24 +1,58 @@
-import { ReactElement } from "react";
+"use client";
+
+import { ReactElement, useState, useEffect } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
 
 import MingcuteSearchLine from "@/icons/MingcuteSearchLine";
 import MingcuteLocationLine from "@/icons/MingcuteLocationLine";
 
 import styles from "./SearchBox.module.css";
 
 export default function SearchBox(): ReactElement {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const initialQuery = searchParams.get("query") || "";
+  const [searchQuery, setSearchQuery] = useState<string>(initialQuery);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    const params = new URLSearchParams(searchParams as any);
+    if (searchQuery.trim()) {
```

Comment:
همون اول trim کنید که بعداً مقداری که Set میکنید هم فاقد Space اضافه باشه

```suggestion
    const params = new URLSearchParams(searchParams as any).trim();
    if (searchQuery) {
```